firecrown/modeling_tools.py

```diff
@@ -29,13 +29,17 @@ def __init__(
         self,
         *,
         pt_calculator: None | pyccl.nl_pt.EulerianPTCalculator = None,
+        hm_calculator: None | pyccl.halos.HMCalculator = None,
+        cM_relation: None | str = None,
         pk_modifiers: None | Collection[PowerspectrumModifier] = None,
         cluster_abundance: None | ClusterAbundance = None,
         ccl_factory: None | CCLFactory = None,
     ):
         super().__init__()
         self.ccl_cosmo: None | pyccl.Cosmology = None
         self.pt_calculator: None | pyccl.nl_pt.EulerianPTCalculator = pt_calculator
+        self.hm_calculator: None | pyccl.halos.HMCalculator = hm_calculator
+        self.cM_relation: None | str = cM_relation
```

Comment:
Why is the c-M relation stored here? Can't this be accessed through the `HMCalculator`? If not, it'd probably be cleaner to change `HMCalculator` such that it is accessible in order to avoid inconsistent states in firecrown. And avoid the need to carry along other things like the mass definition in the future.

Reply:
The halo model calculator does not take a concentration mass relation as an argument, it only accepts mass definition, halo mass/bias functions and some extra parameters related to the integration. The concentration is needed to initialise the IA halo model profile (and other HOD profiles). 

Mass definition, halo mass and bias functions are carried by the halo model calculator. Do you think this is okay then as an implementation then for halo model-based tracers?

Reply:
Right, I guess the c-M relation is more of a halo-profile-specific property, rather than a halo-model property. Would it make sense to have the c-M relation set has part of the systematic? That way different halos (e.g. for baryons) could have different profiles and concentrations. It would also not increase the API surface of `ModellingTools`. What do you think?

Reply:
Yes, it would be better to connect this to the tracer but I could not find a way to have a tracer-specific property that was not `Updatable` and was also a `str`. @marcpaterno Is there a way to do this? For example adding in `__init__` of the `HMAlignmentSystematic` something like
```
self.cM_relation = parameters.register_new_fixed_parameter(
            default_value='Duffy08'
        )
```

Reply:
I'd try adding something like an `ia_hm_cm` field to `WeakLensingArgs`, which gets populated by `HMAlignmentSystematic`. Since this is not a parameter that gets sampled, it can just be an attribute of `HMAlignmentSystematic`. I don't know how such configuration arguments are passed through the various factories though.

Reply:
We want to follow up on improving how this is done, but will merge this PR as-is since it does make a step toward the solution.